web/src/hooks/use-cancellable-promise.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) [2022-2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { useEffect, useRef, useCallback } from "react";
+
+type CancellableWrapper<T> = {
+  /** Cancellable promise */
+  promise: Promise<T>;
+  /** Function for cancelling the promise */
+  cancel: Function;
+};
+
+/**
+ * Creates a wrapper object with a cancellable promise and a function for canceling the promise
+ *
+ * @see useCancellablePromise
+ */
+const makeCancellable = <T>(promise: Promise<T>): CancellableWrapper<T> => {
+  let isCanceled = false;
+
+  const cancellablePromise: Promise<T> = new Promise((resolve, reject) => {
+    promise
+      .then((value) => !isCanceled && resolve(value))
+      .catch((error) => !isCanceled && reject(error));
+  });
+
+  return {
+    promise: cancellablePromise,
+    cancel() {
+      isCanceled = true;
+    },
+  };
+};
+
+/**
+ * Allows using promises in a safer way.
+ *
+ * This hook is useful for safely performing actions that modify a React component after resolving
+ * a promise (e.g., setting the component state once a D-Bus call is answered). Note that nothing
```

Comment:
NP: D-Bus call?

Reply:
Good catch. I simply move this from utils tonits own file, but I can refactor it's documentation as well, althoug not sure if worth it since most probably it will be drop